api/src/Command/Cron.php

```diff
@@ -140,31 +140,43 @@ public function runTask(): bool
         $last_task_timestamp = $this->system->get('last_task_timestamp');
         $last_task_name = $this->system->get('last_task_name');
 
-        // if a task is running and it's not older than MAX_TASK_LOCK_DURATION, do nothing
+        // if a task is running and it's not older than MAX_TASK_LOCK_DURATION, only run our "always" tasks
+        $onlyRunAlwaysTasks = false;
         if ($task_running && isset($last_task_timestamp) && $last_task_timestamp > time() - intval($this->params->get('max_task_lock_duration'))) {
             $this->logger->notice($last_task_name.' is already running since '.(time() - $last_task_timestamp).'s');
 
-            return false;
+            $onlyRunAlwaysTasks = true;
         }
 
-        // if a task is running but it's old, it's probably a lockup.
-        // continue but log it
-        if (isset($last_task_timestamp) && $last_task_timestamp < time() - intval($this->params->get('max_task_lock_duration'))) {
-            $this->logger->notice('Removing old task lock');
-        }
+        // Don't count the always-only task run as the last task when checking for duplicates
+        if (!$onlyRunAlwaysTasks) {
+            // if a task is running but it's old, it's probably a lockup.
+            // continue but log it
+            if (!$onlyRunAlwaysTasks && isset($last_task_timestamp) && $last_task_timestamp < time() - intval($this->params->get('max_task_lock_duration'))) {
+                $this->logger->notice('Removing old task lock');
+            }
 
-        $this->system->set('task_running', true);
+            $this->system->set('task_running', true);
 
-        // get idle hours
-        if (null !== $this->params->get('idle_hours')) {
-            $idle_hours = explode('-', $this->params->get('idle_hours'));
-            $idle_hours[0] = intval($idle_hours[0]);
-            $idle_hours[1] = intval($idle_hours[1]);
-        } else {
-            $idle_hours = [0, 24];
+            // get idle hours
+            if (null !== $this->params->get('idle_hours')) {
+                $idle_hours = explode('-', $this->params->get('idle_hours'));
+                $idle_hours[0] = intval($idle_hours[0]);
+                $idle_hours[1] = intval($idle_hours[1]);
+            } else {
+                $idle_hours = [0, 24];
+            }
         }
 
         foreach ($this->tasks as $task) {
+
+            // If we are only doing "always" tasks, skip others
+            if ($onlyRunAlwaysTasks
+                && isset($task['type'])
```

Comment:
Let's move the `isset` before and make it so that if the type is not defined, we don't run it

Reply:
I think I have updated this to what you're meaning.